auth/src/main/java/com/firebase/ui/auth/util/smartlock/SignInDelegate.java

```diff
@@ -0,0 +1,403 @@
+        super.onActivityResult(requestCode, resultCode, data);
+        if (BuildConfig.DEBUG) {
+            Log.d(TAG, "onActivityResult:" + requestCode + ":" + resultCode + ":" + data);
+        }
+
+        switch (requestCode) {
+            case RC_CREDENTIALS_READ:
+                if (resultCode == RESULT_OK) {
+                    // credential selected from SmartLock, log in with that credential
+                    Credential credential = data.getParcelableExtra(Credential.EXTRA_KEY);
+                    handleCredential(credential);
+
+                    String email = getEmailFromCredential();
+                    String password = getPasswordFromCredential();
+                    if (email != null) {
+                        if (password == null || password.isEmpty()) {
+                            // identifier/provider combination
+                            redirectToIdpSignIn(email, getAccountTypeFromCredential());
+                        } else {
+                            // email/password combination
+                            signInWithEmailAndPassword(email, password);
+                        }
+                    }
+                } else if (resultCode == RESULT_CANCELED
+                        || resultCode == CredentialsApi.ACTIVITY_RESULT_OTHER_ACCOUNT) {
+                    // Smart lock selector cancelled, go to the AuthMethodPicker screen
+                    startAuthMethodChoice();
+                } else if (resultCode == RESULT_FIRST_USER) {
+                    // TODO: (serikb) figure out flow
+                }
+                break;
+            case RC_IDP_SIGNIN:
+            case RC_AUTH_METHOD_PICKER:
+            case RC_EMAIL_FLOW:
+                finish(resultCode, new Intent());
+                break;
+            case RC_PLAY_SERVICES:
+                if (resultCode != RESULT_OK) {
+                    finish(resultCode, new Intent());
+                }
+                break;
+        }
+    }
+
+    private String getEmailFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getId();
+    }
+
+    private String getAccountTypeFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getAccountType();
+    }
+
+    private String getPasswordFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getPassword();
+    }
+
+    private void handleCredential(Credential credential) {
+        mCredential = credential;
+
+        if (IdentityProviders.GOOGLE.equals(credential.getAccountType())) {
+            // Google account, rebuild GoogleApiClient to set account name and then try
+            initGoogleApiClient(credential.getId());
+            // Try silent sign-in with Google Sign In API
+            Auth.GoogleSignInApi.silentSignIn(mGoogleApiClient);
```

Comment:
Before, `googleAutoSignIn` was called both in here and also in `onResult`. In here, it is only called if the condition above is true, but in `onResult` it was called no matter what. I'm pretty sure that was a mistake so I removed the call in `onResult` and I'm only doing it here now.